convert2rhel/utils.py

```diff
@@ -620,4 +627,52 @@ def set_locale():
     os.environ.update({"LC_ALL": "C", "LANG": "C"})
 
 
+def options_from_config_files(config_opts, cfg_path=None):
+    """Get the options from config file.
+
+    The order is:
+    1. if filename is set then load from it
+    2. ~/.convert2rhel.ini
+    3. /etc/convert2rhel.ini
+
+    Return config_opts filled with loaded values.
+    """
+    headers = ["subscription_manager"]  # supported sections in config file
+    config_file = configparser.ConfigParser()
+    paths = CONFIG_PATHS
+    # create dict with all supproted options, all of them set to None
+    # needed for avoiding problems with files priority
+    supproted_opts = {"password": None, "activation_key": None}
+
+    if cfg_path:
+        if not os.path.exists(cfg_path):
+            raise FileNotFoundError
+        paths.insert(0, cfg_path)  # highest priority
+
+    for path in paths:
+        if os.path.exists(path):
+            if oct(os.stat(path).st_mode)[-4:] != "0600":
+                loggerinst.critical("The %s file need to be in mode 0600" % path)
+            config_file.read(path)
+
+            for header in config_file.sections():
+                if header in headers:
+                    for option in config_file.options(header):
+                        if option.lower() in supproted_opts:
+                            # Solving priority
+                            if supproted_opts[option.lower()] is None:
+                                supproted_opts[option] = config_file.get(header, option)
+                                loggerinst.debug("Found %s in %s" % (option, path))
+                        else:
+                            loggerinst.warning("Unsupported option %s in %s" % (option, path))
+                elif header not in headers and header != "DEFAULT":
+                    loggerinst.warning("Unsupported header %s in %s." % (header, path))
+
+    for key, value in supproted_opts.items():
+        if value is not None:
+            setattr(config_opts, key, value)
+
+    return config_opts
```

Comment:
A colleague once told me that good functions should either return the data they've found or set it as a side effect but not both.  I personally favor returning the data and doing the setting in the caller most of the time (That way is easier to unittest, for instance).

So, seeing how the rest of this function operates, I'd change it to not have to pass in config_opts and instead `return supported_opts`.   The caller can then merge `supported_opts` into `config_opts` (Originally I was thinking we'd construct a ToolOpts in here and pass that back but I see from how you've designed this that returning the dict and doing the merging in the caller makes more sense).